Lib/posixpath.py

```diff
@@ -467,10 +472,14 @@ def realpath(filename, *, strict=False):
         else:
             newpath = path + sep + name
         try:
-            st = os.lstat(newpath)
+            st = lstat(newpath)
             if not stat.S_ISLNK(st.st_mode):
                 path = newpath
                 continue
+            if strict and maxlinks != -1:
+                link_count += 1
+                if link_count > maxlinks:
+                    raise OSError(errno.ELOOP, "Too many symbolic links in path", newpath)
```

Comment:
We already can't keep proper track of the symlinks due to caching. e.g. `39->38->...->1->.` 39/39 has 78 symlinks, but we only see 40 of which 39 are read. So, it might make more sense to only count the number of read symlinks.

Reply:
My apologies, we're having the same conversation in two places. Let's continue in #119172?

Reply:
I've addressed this by disabling use of `seen` if `maxlinks` is given. Perhaps you'll be able to find a way to use both simultaneously in your PR!

Reply:
Well, not what I hoped for, but this is probably the cleanest solution. The implementation is already complicated enough.